numpy/compat/py3k.py

```diff
@@ -47,7 +47,12 @@ def asstr(s):
     return str(s)
 
 def isfileobj(f):
-    return isinstance(f, (io.FileIO, io.BufferedReader, io.BufferedWriter))
+    if not isinstance(f, (io.FileIO, io.BufferedReader, io.BufferedWriter)):
+        return False
+    try:
+        return isinstance(f.fileno(), int)
```

Comment:
```suggestion
        # BufferedReader/Writer may raise OSError when
        # fetching `fileno()` (e.g. when wrapping BytesIO).
        f.fileno()
        return True
```
Doesn't really matter, but I don't think we need to check integer and I wanted to rerun that one test anyway.

Thanks for the followups!